samples/snippets/src/main/java/com/example/spanner/PgSpannerSample.java

```diff
@@ -0,0 +1,1603 @@
+      throw (SpannerException) e.getCause();
+    } catch (InterruptedException e) {
+      // Throw when a thread is waiting, sleeping, or otherwise occupied,
+      // and the thread is interrupted, either before or during the activity.
+      throw SpannerExceptionFactory.propagateInterrupt(e);
+    }
+  }
+  // [END spanner_create_index]
+
+  // [START spanner_read_data_with_index]
+  static void readUsingIndex(DatabaseClient dbClient) {
+    try (ResultSet resultSet =
+             dbClient
+                 .singleUse()
+                 .readUsingIndex(
+                     "Albums",
+                     "AlbumsByAlbumTitle",
+                     KeySet.all(),
+                     Arrays.asList("AlbumId", "AlbumTitle"))) {
+      while (resultSet.next()) {
+        System.out.printf("%d %s\n", resultSet.getLong(0), resultSet.getString(1));
+      }
+    }
+  }
+  // [END spanner_read_data_with_index]
+
+  // [START spanner_create_storing_index]
+  static void addStoringIndex(DatabaseAdminClient adminClient, DatabaseId dbId) {
+    OperationFuture<Void, UpdateDatabaseDdlMetadata> op = adminClient.updateDatabaseDdl(
+        dbId.getInstanceId().getInstance(),
+        dbId.getDatabase(),
+        Arrays.asList(
+            "CREATE INDEX AlbumsByAlbumTitle2 ON Albums(AlbumTitle) "
+                + "INCLUDE (MarketingBudget)"),
+        null);
+    try {
+      // Initiate the request which returns an OperationFuture.
+      op.get();
+      System.out.println("Added AlbumsByAlbumTitle2 index");
+    } catch (ExecutionException e) {
+      // If the operation failed during execution, expose the cause.
+      throw (SpannerException) e.getCause();
+    } catch (InterruptedException e) {
+      // Throw when a thread is waiting, sleeping, or otherwise occupied,
+      // and the thread is interrupted, either before or during the activity.
+      throw SpannerExceptionFactory.propagateInterrupt(e);
+    }
+  }
+  // [END spanner_create_storing_index]
+
+  // Before running this example, create a storing index AlbumsByAlbumTitle2 by applying the DDL
+  // statement "CREATE INDEX AlbumsByAlbumTitle2 ON Albums(AlbumTitle) STORING (MarketingBudget)".
+  // [START spanner_read_data_with_storing_index]
+  static void readStoringIndex(DatabaseClient dbClient) {
+    // We can read MarketingBudget also from the index since it stores a copy of MarketingBudget.
+    try (ResultSet resultSet =
+             dbClient
+                 .singleUse()
+                 .readUsingIndex(
+                     "Albums",
+                     "AlbumsByAlbumTitle2",
+                     KeySet.all(),
+                     Arrays.asList("AlbumId", "AlbumTitle", "MarketingBudget"))) {
+      while (resultSet.next()) {
+        System.out.printf(
+            "%d %s %s\n",
+            resultSet.getLong(0),
+            resultSet.getString(1),
+            // TODO: MarketingBudget uppercase not supported, keeping lowercase for now
```

Comment:
Remove TODO before merging. Just using lower case here is OK with me (actually it's the most important part of the sample, as it shows an important difference between Spanner and PostgreSQL).